src/ol/proj/proj.js

```diff
@@ -120,6 +120,9 @@ ol.proj.Projection = function(options) {
    * @type {boolean}
    */
   this.global_ = goog.isDef(options.global) ? options.global : false;
+  if (this.global_) {
+    goog.asserts.assert(!goog.isNull(this.extent_));
```

Comment:
Not sure I completely follow, but it's probably me so can you just ignore me for now :) 

I thought someone may want to use wrapX even though his projection is not "global" (i.e. covering the earth).